homeassistant/components/network/system_health.py

```diff
@@ -0,0 +1,59 @@
+"""Provide info to system health."""
+
+from typing import Any
+
+from homeassistant.components import system_health
+from homeassistant.core import HomeAssistant, callback
+
+from . import Adapter, async_get_adapters, async_get_announce_addresses
+from .models import IPv4ConfiguredAddress, IPv6ConfiguredAddress
+
+
+@callback
+def async_register(
+    hass: HomeAssistant, register: system_health.SystemHealthRegistration
+) -> None:
+    """Register system health callbacks."""
+    register.async_register_info(system_health_info, "/config/network")
+
+
+def _format_ips(ips: list[IPv4ConfiguredAddress] | list[IPv6ConfiguredAddress]) -> str:
+    return ", ".join([f"{ip['address']}/{ip['network_prefix']!s}" for ip in ips])
+
+
+def _get_adapter_ipv4_addresses(adapter: Adapter) -> str:
+    return f"{adapter['name']} ({_format_ips(adapter['ipv4'])})"
+
+
+def _get_adapter_ipv6_addresses(adapter: Adapter) -> str:
+    return f"{adapter['name']} ({_format_ips(adapter['ipv6'])})"
+
+
+def _get_adapter_info(adapter: Adapter) -> str:
+    state = "enabled" if adapter["enabled"] else "disabled"
+    default = ", default" if adapter["default"] else ""
+    auto = ", auto" if adapter["auto"] else ""
+    return f"{adapter['name']} ({state}{default}{auto})"
+
+
+async def _async_get_announce_addresses(hass: HomeAssistant) -> str:
+    return ", ".join(await async_get_announce_addresses(hass))
```

Comment:
I don't think this should be a method. 

Considering it is used once and only contains a single line. Additionally, the place it is used contains similar uses that aren't extracted either.